ingress: Follow-up items for shared LB mode (#21493)
cilium-team/cilium: Documentation/network/servicemesh/ingress.rst

    The ingress controller creates a service of LoadBalancer type, so
    your environment will need to support this.

Cilium allows you to specify load balancer mode for the Ingress resource:

- ``dedicated``: The Ingress controller will create a dedicated loadbalancer
  for the ingress.
- ``shared``: The Ingress controller will use a shared loadbalancer for all

>   Ingress resources.

Reviewer: Would be nice to have a word on the trade-offs, so readers can understand what option is best suited to their use case?

Author: good point :100:

    The ingress controller creates a service of LoadBalancer type, so
    your environment will need to support this.

Cilium allows you to specify load balancer mode for the Ingress resource:

- ``dedicated``: The Ingress controller will create a dedicated loadbalancer
  for the ingress.
- ``shared``: The Ingress controller will use a shared loadbalancer for all
  Ingress resources.

> Each load balancer mode is having its own benefits and drawbacks. The shared
> mode is more efficient use of resources by sharing a single LoadBalancer config

Reviewer: ```suggestion
Each load balancer mode has its own benefits and drawbacks. The shared
mode saves resources by sharing a single LoadBalancer config
```

the cluster, while the dedicated mode can help to avoid blast radius of a single

> Ingress resource affecting others.

Reviewer 2: What is "blast radius"?

I know what the words mean and the general sense, but it feels a bit vague. What kind of problem is it protecting against, is it user error? Cilium bugs? Something else?

Reviewer 2: ^^ In general, wording like "blast radius" is convenient in casual discussions, but in documentation it gives an impression to the reader that the actual behaviour is unknown. However, it is the job of the documentation to describe exactly what the behaviour is, so ideally this should be factual and based on statements of expected behaviour. This can help to assure the reader that the code will work correctly or warn them specifically about which conditions they can guard against by using dedicated mode.

.. Warning::

>     Changing the load balancer mode for an Ingress resource is supported, but
>     there is a risk of losing connectivity to backend services due to new
>     load balancer IP addresses being assigned.

Reviewer 2: In general I would prefer to avoid wording like "is supported" in docs like this, as it's not describing the intended behaviour of the system but it paints a vaguer picture about the relationship between the users and the code.

```suggestion
    It is possible to change the load balancer mode for an Ingress resource.
    When the mode is changed, active connections to backends of the Ingress
    may be terminated during the reconfiguration due to a new load balancer
    IP address being assigned to the Ingress resource.
```

Reviewer 2: ^ Put this way, I wonder if maybe an info callout is more appropriate rather than a warning. I don't think that this documentation is assuming that the user will be changing the load balancer mode at runtime and it's not explicitly telling the user to change this setting.

Warning would be used when there's something pretty unexpected and concerning to call out. I don't think users will be surprised that changing the mode of the Ingress will cause disruption to the traffic to that Ingress.

(I also considered whether it even needs to be a callout or if it could just be regular text in the body of this guide. This seems a bit more like exposition rather than being part of the narrative of the guide, a `.. note::` seems more appropriate.)

Reviewer 2: oh, and I used `for an Ingress resource` above but I think it's a cilium-agent CLI argument? So maybe `for Ingress resources` would be more accurate if it cannot be configured for each Ingress.

Author: thanks a lot :100:.

We have a way to configure LB mode per Ingress resource (i.e. via annotation), hence for an Ingress resource sounds good to me.

    The ingress controller creates a service of LoadBalancer type, so
    your environment will need to support this.

Cilium allows you to specify load balancer mode for the Ingress resource:

- ``dedicated``: The Ingress controller will create a dedicated loadbalancer
  for the ingress.
- ``shared``: The Ingress controller will use a shared loadbalancer for all

>   Ingress resources.

Reviewer 2: nit: Inconsistent capitalization and wording - perhaps we should pick `Ingress resource` and go with that for all cases?
